tools/src/main/java/org/apache/kafka/tools/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,486 @@
+            options = parser.parse(args);
+        } catch (OptionException oe) {
+            CommandLineUtils.printUsageAndExit(parser, oe.getMessage());
+        }
+
+        CommandLineUtils.maybePrintHelpOrVersion(this, "This tool helps to read data from Kafka topics using consumer share groups and outputs it to standard output.");
+
+        checkRequiredArgs();
+
+        Properties consumerPropsFromFile = options.has(consumerConfigOpt)
+                ? Utils.loadProps(options.valueOf(consumerConfigOpt))
+                : new Properties();
+        Properties extraConsumerProps = CommandLineUtils.parseKeyValueArgs(options.valuesOf(consumerPropertyOpt));
+
+        Set<String> groupIdsProvided = checkConsumerGroup(consumerPropsFromFile, extraConsumerProps);
+        consumerProps = buildConsumerProps(consumerPropsFromFile, extraConsumerProps, groupIdsProvided);
+        formatter = buildFormatter();
+    }
+
+    private void checkRequiredArgs() {
+        if (!options.has(topicOpt)) {
+            CommandLineUtils.printUsageAndExit(parser, "--topic is a required argument");
+        }
+        CommandLineUtils.checkRequiredArgs(parser, options, bootstrapServerOpt);
+    }
+
+    private Set<String> checkConsumerGroup(Properties consumerPropsFromFile, Properties extraConsumerProps) {
+        // if the group id is provided in more than place (through different means) all values must be the same
+        Set<String> groupIdsProvided = new HashSet<>();
+        if (options.has(groupIdOpt)) {
+            groupIdsProvided.add(options.valueOf(groupIdOpt));
+        }
+
+        if (consumerPropsFromFile.containsKey(ConsumerConfig.GROUP_ID_CONFIG)) {
+            groupIdsProvided.add((String) consumerPropsFromFile.get(ConsumerConfig.GROUP_ID_CONFIG));
+        }
+
+        if (extraConsumerProps.containsKey(ConsumerConfig.GROUP_ID_CONFIG)) {
+            groupIdsProvided.add(extraConsumerProps.getProperty(ConsumerConfig.GROUP_ID_CONFIG));
+        }
+        if (groupIdsProvided.size() > 1) {
+            CommandLineUtils.printUsageAndExit(parser, "The group ids provided in different places (directly using '--group', "
+                    + "via '--consumer-property', or via '--consumer.config') do not match. "
+                    + "Detected group ids: "
+                    + groupIdsProvided.stream().map(group -> "'" + group + "'").collect(Collectors.joining(", ")));
+        }
+        return groupIdsProvided;
+    }
+
+    private Properties buildConsumerProps(Properties consumerPropsFromFile, Properties extraConsumerProps, Set<String> groupIdsProvided) {
+        Properties consumerProps = new Properties(consumerPropsFromFile);
+        consumerProps.putAll(extraConsumerProps);
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer());
+        if (consumerProps.getProperty(ConsumerConfig.CLIENT_ID_CONFIG) == null) {
+            consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "console-consumer");
```

Comment:
`"console-share-consumer"`. I would expect zero occurrences of "console consumer" or variations in this PR.